packages/RNSocialX/src/components/interaction/video/VideoPlayer.tsx

```diff
@@ -36,143 +36,169 @@ interface IVideoPlayerState {
+							fullscreen={fullScreen}
+							ref={this.playerRef}
+							onFullscreenPlayerDidDismiss={this.onExitFullScreenHandler}
+							onFullscreenPlayerWillPresent={this.onFullScreenPresentHandler}
+							useTextureView={true}
+						/>
+						<VideoControls
+							showPlayButton={showPlayButton}
+							muted={muted}
+							replayVideo={replayVideo}
+							replayVideoText={replayVideoText}
+							resizeToChangeAspectRatio={true}
+							playReady={playReady}
+							thumbOnly={thumbOnly}
+							onVideoPlayStart={this.onVideoPlayStartHandler}
+							onVideoMuteToggle={this.onVideoMuteToggleHandler}
+							onVideoEnterFullScreen={this.onVideoEnterFullScreenHandler}
+							onVideoReplay={this.onVideoReplayHandler}
+						/>
+					</View>
+				</TouchableWithoutFeedback>
+			</VisibleViewPort>
 		);
 	}
 
-	private pauseVideoHandler = () => {
-		this.setState({
-			paused: true,
-		});
+	private checkViewPortVisibleHandler = (isVisible: boolean) => {
+		const { visibleView } = this.state;
+
+		if (isVisible) {
+			if (!visibleView) {
+				this.setState({ paused: false, visibleView: true });
+			}
+		} else {
+			if (visibleView) {
+				this.setState({ paused: true, visibleView: false });
+			}
+		}
+	};
+
+	private onPauseVideoHandler = () => {
+		if (!this.state.ended) {
+			this.setState({
+				paused: true,
+			});
+		}
 	};
 
-	private onVideoMuteToggle = () => {
+	private onVideoMuteToggleHandler = () => {
 		this.setState({
 			muted: !this.state.muted,
 		});
 	};
 
-	private videoReadyHandler = () => {
+	private onVideoReadyHandler = () => {
 		if (!this.state.playReady) {
 			this.setState({
 				playReady: true,
 			});
 		}
 	};
 
-	private onVideoPlayStart = () => {
-		if (this.state.ended && this.playerRef.current) {
-			this.playerRef.current.seek(0);
-		}
+	private onVideoPlayStartHandler = () => {
 		this.setState({
 			ended: false,
 			paused: false,
+			muted: false,
+			replayVideo: false,
 		});
 	};
 
 	private onVideoEndHandler = () => {
 		this.setState({
 			ended: true,
+			replayVideo: true,
+			paused: false,
 		});
 	};
 
-	private onVideoEnterFullScreen = () => {
-		if (this.props.resizeToChangeAspectRatio) {
-			this.setState({
-				resizeMode: this.state.resizeMode === 'cover' ? 'contain' : 'cover',
-			});
+	private onVideoEnterFullScreenHandler = () => {
+		if (Platform.OS === OS_TYPES.Android) {
+			const durationToSeek = 1.0;
+			const uri = this.props.videoURL;
+			this.setState({ paused: true });
+			NativeModules.BridgeModule.showFullscreen(uri, durationToSeek);
 		} else {
-			this.setState({
-				fullScreen: true,
-			});
+			if (this.props.resizeToChangeAspectRatio) {
+				this.setState({
+					resizeMode: this.state.resizeMode === 'cover' ? 'contain' : 'cover',
+				});
+			} else {
+				if (!this.state.replayVideo) {
+					this.setState({ fullScreen: true, muted: false, paused: true });
+				}
+			}
 		}
 	};
 
-	private exitFullScreen = () => {
+	private onExitFullScreenHandler = () => {
```

Comment:
just to note something here, this wont work on android because https://github.com/SocialXNetwork/socialx_react_native/blob/1c851f2ee7f79eccb09e4c00135682440ecc91c0/packages/RNSocialX/android/app/src/main/java/socialx/network/VideoActivity.java#L130 doesnt emit any events indicating a fullscreen exit/hide

Reply:
Yes but on Android, the hide() method is used to hide the Media Player. The function onExitFullScreenHandler() is not working indeed for Android, because it isn't associated with anything on the Native Bridge. But, the exit fullscreen handler has been dealt with in native way.

Reply:
Yep its just to note things up, later on we should emit the exit so we can resume playing the video on the back layer